### PR TITLE
Merge main into doxygen

### PR DESCRIPTION
**Pull request by abheekda1**
@theVerySharpFlat please take a look, some of these have your doxygen changes and manual merging is never fun

### CODE REVIEW

**Comment by theVerySharpFlat** [comment]
This is crazy. Where do I start?

**Comment by abheekda1** [comment]
Any files you've made. Check them and make sure their fine. The Doxygen comments *will* be messed up but you can go back later to fix them. For now at least it builds.

File: lib/include/rmb/motorcontrol/sparkmax/SparkMaxPositionController.h [hunk 1]
```diff
@@ -73,9 +74,12 @@ class SparkMaxPositionController : PositionController<DistanceUnit> {
     rev::SparkMaxPIDController::AccelStrategy accelStrategy;
   };
 
-  /** \deprecated
```
Comment (theVerySharpFlat) [inline lib/include/rmb/motorcontrol/sparkmax/SparkMaxPositionController.h:76]: This shouldn't be deleted. It should be above the constructor and not the struct Follower declaration. Honestly, we need to delete this constructor anyways, so I don't know if we should just go ahead and merge.

Comment (abheekda1) [reply lib/include/rmb/motorcontrol/sparkmax/SparkMaxPositionController.h:76]: Git automerge moment™️

Comment (abheekda1) [reply lib/include/rmb/motorcontrol/sparkmax/SparkMaxPositionController.h:76]: Take a look now, I think it should be fine.

Comment (theVerySharpFlat) [reply lib/include/rmb/motorcontrol/sparkmax/SparkMaxPositionController.h:76]: looks good!

**Comment by abheekda1** [comment]
@theVerySharpFlat Do you believe it's merge worthy now?

**Comment by theVerySharpFlat** [comment]
> @theVerySharpFlat Do you believe it's merge worthy now?

Let's do it

**Pull request merged**